mesonbuild/modules/pkgconfig.py

```diff
@@ -395,13 +407,19 @@ def generate(self, state, args, kwargs):
         deps.add_priv_reqs(kwargs.get('requires_private', []))
         deps.add_cflags(kwargs.get('extra_cflags', []))
 
+        for d in subdirs:
+            if d == '.':
+                deps.add_cflags(['-I${includedir}'])
+            else:
+                deps.add_cflags(self._escape(PurePath('-I${includedir}') / d))
```

Comment:
This changes the order of `add_cflags` now, which will cause `-I${includedir}` to appear *after* `extra_cflags`, which is not what we want.

Reply:
Ack.

Reply:
From taking another pass at this, we add the include dirs and then have a call to 

```deps.add_cflags(kwargs.get('extra_cflags', []))```

So from a logical reading, we'll get ```-I${includedir}``` and then ```extra_cflags``` ?

Reply:
Hm, the test corroborates your review. I'll take another pass at this.

Reply:
according to last comment this should not be resolved conversation.

Reply:
I think I've already fixed this as the tests pass now?

Reply:
The ordering seems correct to me at least in the current version. Wondering if what @nirbheek pointed here was based on previous versions of the PR.

However, you pass it twice in self._escape(), here and in generate_compiler_flags(). That could lead to double quoting of spaces, as far as I understand.

Reply:
I suppose we can drop the escaping in generate_compiler_flags?

Reply:
It is needed in generate_compiler_flags() to escape other paths that come from dependencies. I think you should dop it from here.

Reply:
Done.